src/components/Validator.vue

```diff
@@ -114,19 +125,49 @@ export default {
         this.fileNameAlertMessage = 'Filename is lightshow.fseq';
       }
 
-      if(validation.error) {
-        this.overallAlertType = 'error';
-        this.overallAlertMessage = 'Light Show is invalid';
+      let anyErrorExists = false;
+
+      if (resultsKeys.includes(Duration.toString())) {
+        let durationResults = validation.results[Duration];
+        if (durationResults.isValid) {
+          let durationFormatted = new Date(validation.duration * 1000).toISOString().substr(11, 12);
+          this.durationAlertType = 'success';
+          this.durationAlertMessage = `Found ${validation.frameCount} frames, step time of ${validation.stepTime} ms for a total duration of ${durationFormatted}`;
+        } else {
+          this.durationAlertType = 'error';
+          this.durationAlertMessage = durationResults.message;
+          anyErrorExists = true;
+        }
+      }
 
-        if(validation.memoryUsage > 1) {
+      if (resultsKeys.includes(Memory.toString())) {
+        let memoryResults = validation.results[Memory];
+        if (memoryResults.isValid) {
+          const memoryUsage = parseFloat((validation.memoryUsage * 100).toFixed(2));
+          this.memoryAlertType = 'success';
+          this.memoryAlertMessage = `Used ${memoryUsage}% of the available memory (using ${validation.commandCount} out of 681 allowed commands)`;
+        } else {
           this.memoryAlertType = 'error';
-          this.memoryAlertMessage = validation.error;
+          this.memoryAlertMessage = memoryResults.message;
+          anyErrorExists = true;
         }
+      }
 
-        if(validation.durationSecs > 5 * 60) {
-          this.durationAlertType = 'error';
-          this.durationAlertMessage = validation.error;
+      resultsKeys.forEach(function(key) {
+        if ([FileFormat.toString(), ChannelCount.toString(), FseqType.toString()].includes(key)) {
+          let value = validation.results[key];
+          console.log(value)
+          if (!value.isValid) {
+            this.otherErrorAlertType = 'error';
+            this.otherErrorAlertMessage = value.message;
+            anyErrorExists = true
+          }
         }
+      }, this)
```

Comment:
With the proposed changes on the [the tlsv pr](https://github.com/xsorifc28/tlsv/pull/1), this block of code _should_ be able to be simplified.